src/views/virtualmachines/tree/hooks/useTreeViewData.ts

```diff
@@ -49,11 +57,49 @@ export const useTreeViewData = (): UseTreeViewData => {
     ),
   );
 
+  const [allVMIM] = useKubevirtWatchResource<V1VirtualMachineInstanceMigration[]>({
+    groupVersionKind: VirtualMachineInstanceMigrationModelGroupVersionKind,
+    isList: true,
+    limit: OBJECTS_FETCHING_LIMIT,
+    namespaced: true,
+  });
+
+  const allowedVMIMResources = useK8sWatchResources<{
+    [key: string]: V1VirtualMachineInstanceMigration[];
+  }>(
+    Object.fromEntries(
+      projectNamesLoaded && !isAdmin
+        ? (projectNames || []).map((namespace) => [
+            namespace,
+            {
+              groupVersionKind: VirtualMachineInstanceMigrationModelGroupVersionKind,
+              isList: true,
+              namespace,
+            },
+          ])
+        : [],
+    ),
+  );
+
+  const memoizedVMIMs = useMemo(
+    () =>
+      getLatestMigrationForEachVM(
+        isAdmin
+          ? allVMIM
+          : Object.values(allowedVMIMResources).flatMap((resource) => resource.data),
+      ),
+    [allVMIM, allowedVMIMResources, isAdmin],
+  );
+
+  vmimMapperSignal.value = memoizedVMIMs;
+
   const memoizedVMs = useMemo(
     () => (isAdmin ? allVMs : Object.values(allowedResources).flatMap((resource) => resource.data)),
     [allVMs, allowedResources, isAdmin],
   );
 
+  vmsSignal.value = memoizedVMs;
```

Comment:
what are signals used here for ?

Reply:
Those signals are used in the `TreeViewRightClickAction` component.
When the user right click on the item, from the id element I get VM namespace and name.
With those I search from the `vmsSignal` and `vmimMapperSignal` the VM and relative VMIM to build the action menu.